libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1014,7 +1014,7 @@ bool TypeChecker::visit(TupleExpression const& _tuple)
 						fatalTypeError(components[i]->location(), "Invalid mobile type.");
 
 					if (i == 0)
-						inlineArrayType = types[i];
+						inlineArrayType = types[i]->mobileType();
 					else if (inlineArrayType)
 						inlineArrayType = Type::commonType(inlineArrayType, types[i]);
```

Comment:
Yes, we can do that, but it will be called twice in that case (also inside `commonType`), so I would prefer not calling it.